src/reporting/interfaces/AggregationProperties.ts

```diff
@@ -0,0 +1,415 @@
+     */
+    type: AggregationPropertyType;
+    /**
+     *
+     * @type {AggregationPropertyLinks}
+     * @memberof AggregationProperty
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    _links: AggregationPropertyLinks;
+}
+
+/**
+ * List of AggregationProperties.
+ * @export
+ * @interface AggregationPropertyCollection
+ */
+export interface AggregationPropertyCollection {
+    /**
+     * List of AggregationProperties.
+     * @type {Array<AggregationProperty>}
+     * @memberof AggregationPropertyCollection
+     */
+    aggregationProperties: Array<AggregationProperty>;
+    /**
+     *
+     * @type {PagedResponseLinks}
+     * @memberof AggregationPropertyCollection
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    _links: PagedResponseLinks;
+}
+
+export interface AggregationPropertyLinks {
+    /**
+     *
+     * @type {Link}
+     * @memberof AggregationPropertyLinks
+     */
+    datasource: Link;
+    /**
+     *
+     * @type {Link}
+     * @memberof AggregationPropertyLinks
+     */
+    aggregationTableSet: Link;
+    /**
+     *
+     * @type {Link}
+     * @memberof AggregationPropertyLinks
+     */
+    aggregationTable: Link;
+}
+
+export interface AggregationPropertySingle {
+    /**
+     *
+     * @type {AggregationProperty}
+     * @memberof AggregationPropertySingle
+     */
+    aggregationProperty: AggregationProperty;
+}
+
+export interface AggregationPropertyUpdate {
+    /**
+     * Name of the aggregation property (OData v4 SimpleIdentifier).
+     * @type {string}
+     * @memberof AggregationPropertyUpdate
+     */
+    propertyName?: string;
+    /**
+     * Name of a property in data source table to run the aggregation against.
+     * @type {string}
+     * @memberof AggregationPropertyUpdate
+     */
+    sourcePropertyName?: string;
+    /**
+     * Type of aggregation to run for this property.
+     * @type {string}
+     * @memberof AggregationPropertyUpdate
+     */
+    type?: AggregationPropertyType;
+}
+
+export interface AggregationPropertyCreate {
+    /**
+     * Name of the aggregation property (OData v4 SimpleIdentifier).
+     * @type {string}
+     * @memberof AggregationPropertyCreate
+     */
+    propertyName: string;
+    /**
+     * Name of a property in data source table to run the aggregation against.
+     * @type {string}
+     * @memberof AggregationPropertyCreate
+     */
+    sourcePropertyName: string;
+    /**
+     * Type of aggregation to run for this property.
+     * @type {string}
+     * @memberof AggregationPropertyCreate
+     */
+    type: AggregationPropertyType;
+}
+
+export enum AggregationPropertyType {
+    // eslint-disable-next-line id-blacklist
+    GroupBy = "GroupBy",
+    Count = "Count",
+    Sum = "Sum",
+    Undefined = "Undefined"
+}
+
+/**
+ * Defines an output of Aggregation Table Set.
+ * @export
+ * @interface AggregationTableSet
+ */
+export interface AggregationTableSet {
```

Comment:
@MindaugasButkus Is there a reason datasourceId is not returned?

Reply:
Yes. The full datasource url is returned in _links. You only need to use the `datasourceType` to understand the schema of the result of the GET request with url provided in _links.